### PR TITLE
feat(react-utils): add useGesture hook

### PR DESCRIPTION
**Pull request by Dashice**

### CODE REVIEW

File: packages/react-utils/src/useEvent.tsx
Comment (Dashice) [inline packages/react-utils/src/useEvent.tsx:1]: `.tsx` -> ´.ts`

File: packages/react-utils/src/useGesture.ts [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+  /** The delta in the vertical axis. */
+  deltaY: number
+}
+
+/** The general direction pointer device is travelling over the course of the gesture. */
+export type GestureDirection = 'up' | 'right' | 'down' | 'left' | null
+/** The type of pointer device used for the current gesture. */
+export type GesturePointerType = 'mouse' | 'touch' | 'pen'
+
+export interface GestureStartEvent extends GesturePosition {
```
Comment (Dashice) [inline packages/react-utils/src/useGesture.ts:34]: Add comment to Events themselves.

File: packages/react-utils/src/useGesture.ts [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+  /** Current vertical position in relation to viewport. */
+  y: number
+}
+/** The pointer device rate of travel in both axes. Calculated between move events. */
+export type GestureMovement = {
+  /** The rate of travel in the horizontal axis. */
+  movementX: number
+  /** The rate of travel in the vertical axis. */
+  movementY: number
+}
```
Comment (Dashice) [inline packages/react-utils/src/useGesture.ts:20]: Spacing between comments and types

File: packages/react-utils/src/useGesture.ts [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+    lifespan,
+    onGestureStart,
+    onGestureMove,
+    onGestureEnd,
+  }: GestureOptions = {},
+) => {
+  const startPosition = useRef<GesturePosition>({ x: 0, y: 0 })
+  const startTime = useRef<number | null>(null)
+
+  const previousDelta = useRef<GestureDelta>({ deltaX: 0, deltaY: 0 })
```
Comment (Dashice) [inline packages/react-utils/src/useGesture.ts:170]: Consider combining `previousDelta` and `previousMovement`

File: packages/react-utils/src/useGesture.ts [hunk 4]
```diff
@@ -0,0 +1,306 @@
...
+    movementX: 0,
+    movementY: 0,
+  })
+
+  const isLocked = useRef(true)
+  const offset = useRef<GesturePosition>({ x: 0, y: 0 })
+
+  const previousDirection = useRef<GestureDirection>(null)
+
+  const lifespanTimer = useRef<NodeJS.Timeout | undefined>(undefined)
```
Comment (Dashice) [inline packages/react-utils/src/useGesture.ts:181]: `typeof setTimeout`

**Review by maeertin — APPROVED** [review]
LGTM

**Pull request merged**